### DISCUSSION TITLE
How could this report be improved?

### DISCUSSION THREAD

**Discussion by fridgepoet** [grafana-cold-storage/code-coverage#8] [General]
After a couple of weeks of this workflow being called on cloud data source plugins, this is a place to record some thoughts.

Feedback on existing solution
-----------------------------

-   Difference is interpreted at face value 
    -  "How do I want this to work?" rather than "This positive difference represents well-tested code"
-   Misconception: (+) -> good, (-) or 0 -> bad
    -   0 -> can mean code was already covered
    -   0 -> adding test cases is good but will not increase code coverage
-   Is it misleading to compare percentage against all code?
    -   scale issue (.01) on main Grafana
-  Time saving for PR writing: I usually would write the code coverage on PR description, and now I don't have to :)

Ideas for improvements
----------------------
Convert any agreed-upon ideas to issues

-  https://github.com/grafana/code-coverage/issues/9
-   https://github.com/grafana/code-coverage/discussions/5
-   Use a boolean instead of a number 
    -   (with some fuzzy logic: similar coverage / decreased coverage)
        -   Give supportive feedback about (+) and 0 difference
        -   Give warning feedback about (-) difference
        -   Enlarge range for differences which fall under 0
    -   improved coverage with thresholds to avoid the .01 difference
-  Github interface shows if changed line's code is covered or not (similar to an IDE's code coverage scan)
-  Text color for code coverage report / emoji

@grafana/cloud-datasources

**Comment by andresmgot** [comment]
I think the next more important step in my opinion is to do the report per data source in main Grafana. Apart from the issue that the coverage difference is not that meaningful it seems to be extremely slow. For example, this build is going for an hour now and still going:

![Screenshot from 2022-04-12 11-42-21](https://user-images.githubusercontent.com/4025665/162931778-5ebb9a62-4aed-43ec-900b-3bd04b14e7ac.png)

with that I think we can have a final version.

Having a coverage report with what's covered and what's not would be cool but probably something that we can leave for a v2 if needed.

**Reply by andresmgot** [reply]
also, it seems that we are running the action for backport PRs, we can skip that: https://github.com/grafana/grafana/pull/47650#issuecomment-1096983135
